### PR TITLE
SPARK-22830 Scala Coding style has been improved in Spark Examples

### PR DESCRIPTION
**Pull request by chetkhatri**
## What changes were proposed in this pull request?

* Under Spark Scala Examples: Some of the syntax were written like Java way, It has been re-written as per scala style guide.
* Most of all changes are followed to println() statement.

## How was this patch tested?

Since, All changes proposed are re-writing println statements in scala way, manual run used to test println.

### CODE REVIEW

**Review by HyukjinKwon — COMMENTED** [review]
Not sure if it's worth. String interpolation might be closer to Scala syntax vs I wonder if it makes users to understand it more easily, in particular, who are not used to Scala.

I think I don't have a strong preference between them and I'd just leave them out.

**Comment by chetkhatri** [comment]
@HyukjinKwon I agree with you ! But since Spark is scala project - A lot's of developers refer examples available here and if those are Java Developers they might don't understand that this is right way to do in Scala ! I had a same discussion in Scala Days with developers and I think it does make sense here too.

**Comment by chetkhatri** [comment]
+ @holdenk @sameeragarwal

File: examples/src/main/scala/org/apache/spark/examples/DFSReadWriteTest.scala [hunk 1]
```diff
@@ -97,22 +95,22 @@ object DFSReadWriteTest {
   def main(args: Array[String]): Unit = {
     parseArgs(args)
 
-    println("Performing local word count")
+    println(s"Performing local word count")
```
Comment (srowen) [inline examples/src/main/scala/org/apache/spark/examples/DFSReadWriteTest.scala]: There is no interpolation here; revert changes like this.

File: examples/src/main/scala/org/apache/spark/examples/DFSReadWriteTest.scala [hunk 2]
```diff
@@ -127,11 +125,11 @@ object DFSReadWriteTest {
     spark.stop()
 
     if (localWordCount == dfsWordCount) {
-      println(s"Success! Local Word Count ($localWordCount) " +
-        s"and DFS Word Count ($dfsWordCount) agree.")
+      println(s"Success! Local Word Count ($localWordCount) 
+                and DFS Word Count ($dfsWordCount) agree.")
```
Comment (srowen) [inline examples/src/main/scala/org/apache/spark/examples/DFSReadWriteTest.scala]: This should be reverted; you've added a bunch of space to the string.

File: examples/src/main/scala/org/apache/spark/examples/LocalFileLR.scala [hunk 3]
```diff
@@ -58,10 +58,10 @@ object LocalFileLR {
 
     // Initialize w to a random value
     val w = DenseVector.fill(D) {2 * rand.nextDouble - 1}
-    println("Initial w: " + w)
+    println(s"Initial w: ${w}")
```
Comment (srowen) [inline examples/src/main/scala/org/apache/spark/examples/LocalFileLR.scala]: You can just write `$w` in cases like this

**Review by srowen — CHANGES_REQUESTED** [review]

File: examples/src/main/scala/org/apache/spark/examples/BroadcastTest.scala [hunk 4]
```diff
@@ -42,7 +42,7 @@ object BroadcastTest {
     val arr1 = (0 until num).toArray
 
     for (i <- 0 until 3) {
-      println("Iteration " + i)
+      println(s"Iteration ${i}")
```
Comment (markhamstra) [inline examples/src/main/scala/org/apache/spark/examples/BroadcastTest.scala]: Beyond the unnecessary { } that @srowen has already mentioned, this isn't really a style improvement. `"a string " + anotherString` is arguably at least as good stylistically as using string interpolation for such simple concatenations of a string reference to the end of a string literal. It's only when there are multiple concatenations and/or multiple string references that interpolation is clearly the better way.

Comment (chetkhatri) [reply examples/src/main/scala/org/apache/spark/examples/BroadcastTest.scala]: @markhamstra Thank you for valueable suggestion, I am addressed and did new commit.

File: examples/src/main/scala/org/apache/spark/examples/DFSReadWriteTest.scala (hunk 1 above)
Comment (chetkhatri) [reply examples/src/main/scala/org/apache/spark/examples/DFSReadWriteTest.scala]: @srowen Thanks for review, I did addressed changes. Please review

File: examples/src/main/scala/org/apache/spark/examples/DFSReadWriteTest.scala [hunk 5]
```diff
@@ -49,12 +49,10 @@ object DFSReadWriteTest {
   }
 
   private def printUsage(): Unit = {
-    val usage: String = "DFS Read-Write Test\n" +
-    "\n" +
-    "Usage: localFile dfsDir\n" +
-    "\n" +
-    "localFile - (string) local file to use in test\n" +
-    "dfsDir - (string) DFS directory for read/write tests\n"
+    val usage = s"""DFS Read-Write Test
```
Comment (mgaido91) [inline examples/src/main/scala/org/apache/spark/examples/DFSReadWriteTest.scala]: here you should use
```
"""
 |...
 """.stripMargin
```
otherwise you introduce a lot of spaces

File: examples/src/main/scala/org/apache/spark/examples/LocalALS.scala [hunk 6]
```diff
@@ -95,7 +95,7 @@ object LocalALS {
 
   def showWarning() {
     System.err.println(
-      """WARN: This is a naive implementation of ALS and is given as an example!
+      s"""WARN: This is a naive implementation of ALS and is given as an example!
```
Comment (mgaido91) [inline examples/src/main/scala/org/apache/spark/examples/LocalALS.scala]: please here revert the change and do the same in all similar places, since there is no variable to interpolate

File: examples/src/main/scala/org/apache/spark/examples/HdfsTest.scala [hunk 7]
```diff
@@ -39,7 +39,7 @@ object HdfsTest {
       val start = System.currentTimeMillis()
       for (x <- mapped) { x + 2 }
       val end = System.currentTimeMillis()
-      println("Iteration " + iter + " took " + (end-start) + " ms")
+      println(s"Iteration ${iter} took ${(end-start)} ms")
```
Comment (HyukjinKwon) [inline examples/src/main/scala/org/apache/spark/examples/HdfsTest.scala]: Let's just write as `$iter` and `$end-start`

File: examples/src/main/scala/org/apache/spark/examples/SparkALS.scala [hunk 8]
```diff
@@ -80,7 +80,7 @@ object SparkALS {
 
   def showWarning() {
     System.err.println(
-      """WARN: This is a naive implementation of ALS and is given as an example!
+      s"""WARN: This is a naive implementation of ALS and is given as an example!
```
Comment (HyukjinKwon) [inline examples/src/main/scala/org/apache/spark/examples/SparkALS.scala]: Seems we don't need `s`.

File: examples/src/main/scala/org/apache/spark/examples/SparkALS.scala [hunk 9]
```diff
@@ -100,7 +100,7 @@ object SparkALS {
         ITERATIONS = iters.getOrElse("5").toInt
         slices = slices_.getOrElse("2").toInt
       case _ =>
-        System.err.println("Usage: SparkALS [M] [U] [F] [iters] [partitions]")
+        System.err.println(s"Usage: SparkALS [M] [U] [F] [iters] [partitions]")
```
Comment (HyukjinKwon) [inline examples/src/main/scala/org/apache/spark/examples/SparkALS.scala]: ditto for `s`.

**Comment by chetkhatri** [comment]
@HyukjinKwon @mgaido91 @srowen All the changes are addressed and committed, please do review and needful.

**Review by srowen — APPROVED** [review]

**Comment by chetkhatri** [comment]
@srowen why only recent commit is going to merge, can't we get "squash merge" ? 
Please re-run test build. and let me know if still seems wrong.

**Comment by srowen** [comment]
@chetkhatri not sure what you mean. The whole change here is considered, and doesn't pass style checks. Look at the test results and fix the errors.

**Comment by chetkhatri** [comment]
@srowen Thanks for response, correct - i went through error of jenkins and found error Online[https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/4018/console] which fixed accordingly to me and committed so please take a look, if not correct please suggest the same.
Thank you

**Comment by chetkhatri** [comment]
@srowen Thank you for re-run, now it passes all.

**Comment by chetkhatri** [comment]
@srowen I think, we can merge this now.

**Comment by srowen** [comment]
Merged to master
